server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -871,15 +871,12 @@ public static String findV2Template(Metadata metadata, String indexName, boolean
     }
 
     /**
-     * Resolve the given v2 template into an ordered list of mappings
+     * Collect the given v2 template into an ordered list of mappings.
      */
-    public static List<CompressedXContent> resolveMappings(final ClusterState state, final String templateName) {
+    public static List<CompressedXContent> collectMappings(final ClusterState state, final String templateName) {
         final ComposableIndexTemplate template = state.metadata().templatesV2().get(templateName);
         assert template != null : "attempted to resolve mappings for a template [" + templateName +
             "] that did not exist in the cluster state";
-        if (template == null) {
-            return List.of();
-        }
```

Comment:
I think I'd rather keep this guard than throw a null pointer exception a couple of lines down, is there a reason you removed it?

Reply:
or at the least, convert this to a real exception rather than an NPE if you want to get rid of it

Reply:
We assert right above that `template != null`, so I thought it'd be safe to simplify this.

Reply:
I think even with the assertion I'd rather have a helpful message for production clusters than a `NullPointerException` (since assertions are off by default except in tests)

Reply:
I'll just revert this, I don't feel strongly about it!